packages/firestore/test/unit/lite-api/types.test.ts

```diff
@@ -695,4 +695,79 @@ describe('FirestoreTypeConverter', () => {
       expect(data.numberProperty).to.equal(43);
     }
   });
+
+  it('setDoc() fails to compile if `data` argument is missing properties', () => {
+    async function _(docRef: DocumentReference): Promise<void> {
+      const converter = new ThrowingConverter<
+        { foo: string },
+        { bar: number }
+      >();
+      const docRefWithConverter = docRef.withConverter(converter);
+      // @ts-expect-error `data` argument is missing `foo` property.
+      await setDoc(docRefWithConverter, { bar: 42 });
+    }
+  });
+
+  it('setDoc() fails to compile if `data` argument has incorrect type for a property', () => {
+    async function _(docRef: DocumentReference): Promise<void> {
+      const converter = new ThrowingConverter<{ foo: string }, {}>();
+      const docRefWithConverter = docRef.withConverter(converter);
+      // @ts-expect-error The `data` argument has the wrong type for `foo`.
+      await setDoc(docRefWithConverter, { foo: 42 });
+    }
+  });
+
+  it('updateDoc() fails to compile if `data` argument is missing properties', () => {
+    async function _(docRef: DocumentReference): Promise<void> {
+      const converter = new ThrowingConverter<
+        { foo: string },
+        { bar: number }
+      >();
+      const docRefWithConverter = docRef.withConverter(converter);
+      // @ts-expect-error `data` argument is missing `bar` property.
+      await updateDoc(docRefWithConverter, { foo: 'foo' });
+    }
+  });
+
+  it('updateDoc() fails to compile if `data` argument has incorrect type for a property', () => {
+    async function _(docRef: DocumentReference): Promise<void> {
+      const converter = new ThrowingConverter<{}, { bar: number }>();
+      const docRefWithConverter = docRef.withConverter(converter);
+      // @ts-expect-error The `data` argument has the wrong type for `bar`.
+      await updateDoc(docRefWithConverter, { bar: 'bar' });
+    }
+  });
+
+  it('getDoc() returns AppModelType', () => {
+    async function _(docRef: DocumentReference): Promise<void> {
+      const converter = new ThrowingConverter<
+        { foo: string },
+        { bar: number }
+      >();
+      const docRefWithConverter = docRef.withConverter(converter);
+      const snapshot = await getDoc(docRefWithConverter);
+      const data: { foo: string } = snapshot.data()!;
+      expect(data.foo).to.equal('foo');
```

Comment:
FWIW. This line confused me. I was trying to figure out how the value could equal `'foo'`, or anything, when using a throwing converter. But then I realized the code is never executed. Might confuse a future reader as well.